account_payment_order/models/account_payment.py

```diff
@@ -49,6 +50,14 @@ def _compute_payment_method_line_fields(self):
                 )
         return res
 
+    @api.depends("payment_line_ids", "payment_line_ids.date")
+    def _compute_payment_line_date(self):
+        for item in self:
+            if item.payment_line_ids:
+                item.payment_line_date = item.payment_line_ids[:1].date
+            else:
+                item.payment_line_date = item.payment_line_date
```

Comment:
I think there's no need for the `if`:

```suggestion
        for item in self:
            item.payment_line_date = item.payment_line_ids[:1].date
```